distributed/shuffle/tests/test_shuffle.py

```diff
@@ -889,29 +803,20 @@ async def test_tail(c, s, a, b):
     await clean_scheduler(s)
 
 
-@pytest.mark.xfail(reason="Tombstone prohibits multiple calls to head")
-@gen_cluster(client=True, nthreads=[("127.0.0.1", 4)] * 2)
+@gen_cluster(client=True)
 async def test_repeat(c, s, a, b):
     df = dask.datasets.timeseries(
         start="2000-01-01",
         end="2000-01-10",
         dtypes={"x": float, "y": float},
         freq="100 s",
     )
-    out = dd.shuffle.shuffle(df, "x", shuffle="p2p")
-    await c.compute(out.head(compute=False))
-
-    await clean_worker(a, timeout=2)
-    await clean_worker(b, timeout=2)
-    await clean_scheduler(s, timeout=2)
-
-    await c.compute(out.tail(compute=False))
+    await c.compute(dd.shuffle.shuffle(df, "x", shuffle="p2p"))
 
-    await clean_worker(a, timeout=2)
-    await clean_worker(b, timeout=2)
-    await clean_scheduler(s, timeout=2)
+    while s.tasks:
+        await asyncio.sleep(0)
 
-    await c.compute(out.head(compute=False))
+    await c.compute(dd.shuffle.shuffle(df, "x", shuffle="p2p"))
```

Comment:
I think we should have another test that actually computes the identical graph, i.e.

```python
graph = dd.shuffle.shuffle(df, ...).size
await c.compute(graph)
await asyncio.sleep(0.1
await c.compute(graph)
```

If you are generating the graph from scratch, there is otherwise the possibility that some ID is regenerated. The fact that the ID is currently deterministic is more or less an implementation detail but I wouldn't want to hold on to this, at least not for the shuffle layer

Reply:
Fair point, I'll add that.

Reply:
See `test_repeat_shuffle_operation` and `test_repeat_shuffle_instance`